### PR TITLE
Add basic server-side apply test to test-cmd

### PR DESCRIPTION
**Pull request by apelisse**
**What type of PR is this?**
/kind cleanup

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @jennybuckley @kwiesmueller

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: hack/make-rules/test-cmd.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ function run_kube_apiserver() {
   AUTHORIZATION_MODE="RBAC,AlwaysAllow"
 
   # Enable features
-  ENABLE_FEATURE_GATES="DryRun=true"
+  ENABLE_FEATURE_GATES="ServerSideApply=true"
```
Comment (spiffxp) [inline hack/make-rules/test-cmd.sh]: I think this is causing `run_kubectl_run_tests` to flake

**Comment by apelisse** [comment]
That’s great to know, do you have an example failure of such a flake?
Thanks!

**Comment by spiffxp** [comment]
I have a few occurrences of the flake linked in the linked issue... I'm trying to repro and bisect locally, and I'm not yet certain it's this PR, I'll keep you posted

**Comment by apelisse** [comment]
Thanks, I missed it from my phone. Trying to look at it too.

File: hack/make-rules/test-cmd.sh (hunk 1 above)
Comment (spiffxp) [reply hack/make-rules/test-cmd.sh]: disregard, it works fine, I'm off to bisect beyond this PR

**Pull request merged**